roomserver/internal/input/input_events.go

```diff
@@ -193,9 +193,9 @@ func (r *Inputer) processRoomEvent(
 			serverRes.ServerNames = append(serverRes.ServerNames, input.Origin)
 			delete(servers, input.Origin)
 		}
-		if senderDomain != input.Origin && senderDomain != r.Cfg.Matrix.ServerName {
-			serverRes.ServerNames = append(serverRes.ServerNames, senderDomain)
-			delete(servers, senderDomain)
+		if sender.Domain() != input.Origin && sender.Domain() != r.Cfg.Matrix.ServerName {
+			serverRes.ServerNames = append(serverRes.ServerNames, sender.Domain())
+			delete(servers, sender.Domain())
```

Comment:
We should just optimistically do this if `sender` can be resolved (i.e we know an `mxid_mapping` for this sender key). We should NOT be failing to process the event if we have no `mxid_mapping`.